app/javascript/App.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { fetchUserData } from './redux/user/userSlice.js';
+import Great from './components/great.js';
+
+ const App = () => {
+    const dispatch = useDispatch();
+    useEffect(() => {
+      dispatch(fetchUserData());
+    }, [dispatch]);
+  return (<Great/>);
+}
```

Comment:
- [ ] As the previous reviewer mentioned, as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20the%20App-,component%20with%20react%2Drouter%20and%20render%20it%20in%20your%20static%20view.,-Create%20the%20Greeting), Please add react-router to the app component to make the app serve as single page application.

[Hint] Install `react-router-dom` library then make the app component like this:

```javascript
import React from "react";
import Great from "./components/great.js";
import { Route, BrowserRouter as Router, Routes } from "react-router-dom";

const App = () => {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Great />} />
      </Routes>
    </Router>
  );
};

export default App;
```

- [ ] [Optional] [Highly Recommended] It would be better professionally if you move the dispatch of the fetch user data action to the great component to serve the single responsibility principle.

[Hint] The Great component will look like this:

```javascript
import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import {fetchUserData} from '../redux/user/userSlice'
 
export const Great = () => {
  const greeting = useSelector((state) => state.user.data.greeting)

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchUserData());
  }, []);

  return(
    <h1>{greeting}</h1>
  );
}

export default Great;
```